app/src/main/java/unipd/se18/ocrcamera/GalleryActivity.java

```diff
@@ -0,0 +1,385 @@
+    //Code for internet permission
+    private static final int REQUEST_PERMISSION_CODE = 500;
+    private FragmentManager fm;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_gallery);
+        fm = getSupportFragmentManager();
+
+        //When the activity starts the first thing that I do i check for storage permission
+        //If the function returns true I'll load the home page.
+        if(verifyStoragePermission())
+        {
+            loadHomeFragment();
+        }
+    }
+
+    /**
+     * Function for load the home fragment from onActivityCreated and
+     * onRequestPermissionsResult in case I get the storage permission from the permission
+     */
+    private void loadHomeFragment()
+    {
+        FragmentTransaction ft = fm.beginTransaction();
+        ft.add(R.id.fragmentPlaceHolder, new MainFragment(),getString(R.string.homeFragmentTag));
+        ft.commit();
+    }
+
+    @Override
+    public void onBackPressed()
+    {
+        int backStackEntryCount = getSupportFragmentManager().getBackStackEntryCount();
+        //If there is more than 1 fragment running I close the
+        //current one (DetailsFragment) and return to the prevous (HomeFragment)
+        //Or I just call the default onBackPressed which closes the activity
+        if (backStackEntryCount != 0) {
+            fm.popBackStack();
+        } else {
+            super.onBackPressed();
+        }
+    }
+
+
+
+    /************************************************************/
+    /***************************FRAGMENTS************************/
+    /************************************************************/
+
+
+    /**
+     * Fragment of the gallery layout
+     * @author Romanello Stefano
+     */
+    public static class MainFragment extends Fragment
+    {
+        GalleryManager.RecycleCardsAdapter cardAdapter;
+        /**
+         * Event triggered once everything in the activity have finished loading
+         */
+        @Override
+        public void onActivityCreated(Bundle savedInstanceState)
+        {
+            super.onActivityCreated(savedInstanceState);
+            ActionBar actionBar =((GalleryActivity)getActivity()).getSupportActionBar();
+            actionBar.setTitle(R.string.galleryFragmentTitle);
+            actionBar.setDisplayHomeAsUpEnabled(false);
+            actionBar.setHomeButtonEnabled(false);
+        }
+
+        /**
+         * Event triggered when I create the view
+         */
+        @Override
+        public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState)
+        {
+            // Defines the xml file for the fragment
+            return inflater.inflate(R.layout.activity_gallery_content, parent, false);
+        }
+
+        /**
+         * This event is triggered soon after onCreateView().
+         * Any view setup should occur here.
+         */
```

Comment:
Could more useful than this be describing what things do this method? When it is launched is in the Android documentations and it is possible putting a link about that.